llmfoundry/callbacks/loss_perp_v_len_callback.py

```diff
@@ -0,0 +1,349 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, Mapping, Optional, Tuple
+
+import torch
+from composer.core import Callback, State
+from composer.loggers import Logger, MLFlowLogger
+from composer.utils import dist
+from torchmetrics import Metric
+
+from llmfoundry.models.mpt import ComposerMPTCausalLM
+
+__all__ = [
+    'LossPerpVsContextLengthLogger',
+]
+
+
+class LossPerpVsContextLengthLogger(Callback):
+    """Logs the average loss and perplexity for every context length.
+
+    Args:
+        log_batch_interval (int): The interval for logging.
+        compute_batch_interval (int): The interval for computing the metric.
+        ignore_index (int): Specifies a target value that is ignored for computing loss.
+    """
+
+    def __init__(
+        self,
+        log_batch_interval: int,
+        compute_batch_interval: int,
```

Comment:
Should we add a comment here saying that eventually we should consider making this into one arg?
(once that other issue gets resolved)

Reply:
Done.